lua/entities/gmod_wire_coldetector.lua

```diff
@@ -0,0 +1,247 @@
+AddCSLuaFile()
+
+if !WireLib then return end
+DEFINE_BASECLASS( "base_wire_entity" )
+ENT.PrintName       = "Wire Collision Detector"
+ENT.WireDebugName	= "Collision Detector"
+local f =""
+local i = 1
+/*if CLIENT then
+	while(true) do
+		name, value = debug.getupvalue(effects.Create,i)
+		if name == nil then break end
+		if name == "EffectList" then
+			PrintTable(value)
+		end
+		i=i+1
+	end
+end*/
+
+if CLIENT then return end
+local stickplayer=CreateConVar( "wire_col_detector_stickplayer", 0, 0, "Allow linked entity stick to players" )
+local stickprops=CreateConVar( "wire_col_detector_stickprops", 0, 0, "Allow sticking to other players stuff" )
+function entitiesCollide(ent,data)
+	if !IsValid(ent.ColSensor) || ent.ColSensor.LinkedEnt != ent then return end
+	ent.ColSensor:ApplyCollide(data,ent)
+end
+
+function ENT:Initialize()
+	self:PhysicsInit( SOLID_VPHYSICS )
+	self:SetMoveType( MOVETYPE_VPHYSICS )
+	self:SetSolid( SOLID_VPHYSICS )
+	self.LinkedEnt=self
+	self.Inputs = WireLib.CreateInputs( self, { "Stick to world", "Stick to props", "Stick to npc/player"} )
+	self.Outputs = WireLib.CreateOutputs(self, {"Collided","Speed","Delta Time","Hit Position [VECTOR]","Hit Normal [VECTOR]","Velocity [VECTOR]","Hit Entity Velocity [VECTOR]","Hit Entity [ENTITY]","Entity Position [VECTOR]","Entity Angle [ANGLE]","Hit Entity Position [VECTOR]","Hit Entity Angle [ANGLE]"})
+	//self:Setup(1,1,false,false,0,0,0,0,0,0,false,"","","",0,0,0,false,false,0,0,false,false,false,false)
+end
+function ENT:Setup(AllowConstrained,NoPhysics)
+	self.AllowConstrained=AllowConstrained
+	self.NoPhysics=NoPhysics
+	self:ShowOutput()
+end
+
+function ENT:PhysicsCollide(data,collider)
+	if self.LinkedEnt==self then
+		self:ApplyCollide(data,self)
+	end
+end
+
+function ENT:ApplyCollide(data,ent)
+	if !self.AllowConstrained and table.HasValue(constraint.GetAllConstrainedEntities(ent),data.HitEntity) then return end
+	local posxd=ent:GetPhysicsObject():GetPos()
+	local anglesxd=ent:GetPhysicsObject():GetAngles()
+	if data.HitEntity==nil then return end
+	local ent2pos=data.HitEntity:GetPos()
+	local ent2ang=data.HitEntity:GetAngles()
+	//print(gamemode.Call("CanProperty",self:GetPlayer(),"weld",data.HitEntity))
+	if !(GetConVarNumber("wire_col_detector_stickplayer")==0 && IsValid(data.HitEntity) && data.HitEntity:IsPlayer())&&
```

Comment:
Clean up this mess please, this is horrible!
1. Add spaces around binary operators
2. Extract some of this mess out into functions
3. Extract the if condition into a function
4. Clean it up by shortcutting [1] some of these conditions

And make sure you use standard lua where possible
"not", "and" and "or" instead of "!", "&&" and "||"

[1] By shortcutting I mean turn ``return a or (b and c)`` into this:
````lua
if a then return true end
return b and c
````
and further into this:
````lua
if a then return true end
if not b a then return false end
return c
````
You do that until you have nice readable code where you don't have to hurt your head just to understand what it does.